### DISCUSSION TITLE
Save user session

### DISCUSSION THREAD

**Discussion by AzimovIz** [cloud-py-api/nc_py_api#275] [Ideas]
I am writing an application to integrate nextcloud into another application. I want the user to be able to log in once and use this login until it expires. I have not found a way to "save the session". 
For example, it might look like this:
```python
# login
nc = Nextcloud(nextcloud_url="mynextcloud.com", nc_auth_user="admin", nc_auth_pass="admin")
# save session
nc.save_session(filepath)

# load session аfter some time (maybe you should probably add a validation check)
nc = Nextcloud.from_session(filepath)
```
How hard is it to implement this?

*I cannot save my username and password in order to log in again every time, as this is contrary to the security policy in the place where it will be used.

**Comment by bigcat88** [comment]
Do you just need convenient functions for saving cookies and loading them?

**Reply by AzimovIz** [reply]
Maybe this is what I need. I want the user's session to be loaded from a file/string/other
I will use some kind of database to store sessions, if it is a text string, it will be great.

**Reply by AzimovIz** [reply]
I'm sorry for my English, I don't speak it and use a translator.

**Reply by bigcat88** [reply]
will try to add that a little bit later

**Reply by AzimovIz** [reply]
Thank you so much for doing this project!

**Comment by bigcat88** [comment]
it is not so easy, as such changes to server is required:

https://github.com/nextcloud/server/pull/46760

And they will not be backported to previous versions, so Nextcloud 30 should be minimun supported version for this.

---

Maybe in future we will try to add this to nc-py-api in client mode, but now we have time only to develop it as a helper library for the AppAPI Ecosystem.

Some stuff are equal for both modes, like new webhooks for NC30 and they will be avalaible in usual client mode, but these speciak `cookies` things are not priority as they are not avalaible for ExApps currently.

If someone wants to do this only for "client" mode, we are willing to accepts and review PR's regarding this.

**Reply by AzimovIz** [reply]
Hi, thanks for the reply. Unfortunately, I am not good enough to make a qualitative contribution. However, I will try to do something.
